connectivity/drivers/lora/TARGET_STM32WL/STM32WL_LoRaRadio.cpp

```diff
@@ -0,0 +1,1561 @@
+    { 46900, 0x14 },
+    { 58600, 0x0C },
+    { 78200, 0x1B },
+    { 93800, 0x13 },
+    { 117300, 0x0B },
+    { 156200, 0x1A },
+    { 187200, 0x12 },
+    { 234300, 0x0A },
+    { 312000, 0x19 },
+    { 373600, 0x11 },
+    { 467000, 0x09 },
+    { 500000, 0x00 }, // Invalid Bandwidth
+};
+
+const uint8_t sync_word[] = {0xC1, 0x94, 0xC1, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+// in ms                                 SF12    SF11    SF10    SF9    SF8    SF7
+const float lora_symbol_time[3][6] = {{ 32.768, 16.384, 8.192, 4.096, 2.048, 1.024 },  // 125 KHz
+    { 16.384, 8.192,  4.096, 2.048, 1.024, 0.512 },  // 250 KHz
+    { 8.192,  4.096,  2.048, 1.024, 0.512, 0.256 }
+}; // 500 KHz
+
+STM32WL_LoRaRadio::STM32WL_LoRaRadio(PinName rf_switch_ctrl1,
+									 PinName rf_switch_ctrl2,
+									 PinName rf_switch_ctrl3)
+    :
+        _rf_switch_ctrl1(rf_switch_ctrl1, PIN_OUTPUT, PullNone, 0),
+        _rf_switch_ctrl2(rf_switch_ctrl2, PIN_OUTPUT, PullNone, 0),
+        _rf_switch_ctrl3(rf_switch_ctrl3, PIN_OUTPUT, PullNone, 0)
+#ifdef MBED_CONF_RTOS_PRESENT
+    , irq_thread(osPriorityRealtime, 1024, NULL, "Thread_STM32WL")
+#endif
+		
+{
+    _radio_events = NULL;
+    _image_calibrated = false;
+    _force_image_calibration = false;
+    _active_modem = MODEM_LORA;
+#ifdef MBED_CONF_RTOS_PRESENT
+    irq_thread.start(callback(this, &STM32WL_LoRaRadio::rf_irq_task));
+#endif
+}
+
+STM32WL_LoRaRadio::~STM32WL_LoRaRadio()
+{
+
+}
+
+/**
+ * Acquire lock
+ */
+void STM32WL_LoRaRadio::lock(void)
+{
+    mutex.lock();
+}
+
+/**
+ * Release lock
+ */
+void STM32WL_LoRaRadio::unlock(void)
+{
+    mutex.unlock();
+}
+
+#ifdef MBED_CONF_RTOS_PRESENT
+/**
+ * Thread task handling IRQs
+ */
+void STM32WL_LoRaRadio::rf_irq_task(void)
+{
+    for (;;) {
+        uint32_t flags = ThisThread::flags_wait_any(0x7FFFFFFF);
+
+        lock();
+        if (flags & SIG_INTERRUPT) {
+            irq_process_isr();
+        }
+        unlock();
+    }
+}
+#endif
+
+void STM32WL_LoRaRadio::error_handler(HAL_StatusTypeDef error)
+{
+  printf ("ERROR in STM32WL_loRaRadio.c : %d", error);
+  /* USER CODE BEGIN error_handler_Debug */
+  /* User can add his own implementation to report the HAL error return state */
+  while (1)
+  {
+  }
+  /* USER CODE END error_handler_Debug */
+}
+
+void STM32WL_LoRaRadio::irq_process_isr()
+{
+#ifdef MBED_CONF_RTOS_PRESENT
+    irq_thread.flags_set(SIG_INTERRUPT);
+#else
+    RadioIrqProcess();
+#endif
+}
+
+uint16_t STM32WL_LoRaRadio::get_irq_status(void)
+{
+    uint8_t status[2];
+
+    read_opmode_command((uint8_t) RADIO_GET_IRQSTATUS, status, 2);
+    return (status[0] << 8) | status[1];
+}
+
+void STM32WL_LoRaRadio::clear_irq_status(uint16_t irq)
+{
+    uint8_t buf[2];
+
+    buf[0] = (uint8_t)(((uint16_t) irq >> 8) & 0x00FF);
+    buf[1] = (uint8_t)((uint16_t) irq & 0x00FF);
+    write_opmode_command((uint8_t) RADIO_CLR_IRQSTATUS, buf, 2);
+}
+
+//TODO - Better do CAD here. CAD code is already part of the driver
```

Comment:
TODO to be handled

Reply:
comment from sx126x radio implementation: removed